dulwich/index.py

```diff
@@ -495,6 +495,27 @@ def has_conflicts(self) -> bool:
                 return True
         return False
 
+    def set_merge_conflict(self, base_entry, base_time, this_entry, this_time, other_entry, other_time):
```

Comment:
please add types

Reply:
why does this take a separate time argument? Couldn't the caller just set that if they needed to?

Reply:
please add tests for this new method